### ISSUE TITLE
Geneformer and scGPT embeddings

### ISSUE THREAD

**Issue by HesamAsad** [valence-labs/Tx-Evaluation#2]
Hi,

I was wondering how you extracted the foundation model embeddings for the L1000 dataset. To the best of my understanding, Geneformer requires raw gene expression counts, which it then normalizes using pre-computed median values (calculated from single-cell data, not bulk). After normalization, the data is ranked, tokenized, and subsequently embedded.

However, the L1000 dataset you've uploaded contains normalized values rather than raw counts. Could you please clarify your approach for generating the SCGPT and Geneformer embeddings for this dataset? If possible, I would greatly appreciate access to any notebooks or scripts used, as I’m interested in reproducing the results.

Thank you!

**Comment by rahul-ohlan** [comment]
Hi @IhabBendidi @HesamAsad 

I had the same question regarding extracting embeddings for the L1000 dataset from Geneformer. 
The paper mentions: 

> For training the scVI model, we configure it with a latent‐space size of 256 and set the hidden‐space dimension to nhidden with two layers. The number of highly variable genes selected differs by dataset: 15 000 genes for the Joung overexpression dataset, 8 000 genes for the Replogle dataset and 12 000 genes for the L1000 assay. The dispersion parameter is set to “gene”, and the latent distribution is normal. We test three gene-likelihood models—zero-inflated negative binomial (ZINB), negative binomial (NB) and Poisson. To ensure consistent comparison, **we normalise every sample to a library size of 10 000 counts and apply a log transformation before training scVI; this normalisation mitigates library-size effects and improves model stability.**

I assume same transformation is required for training Geneformer and scGPT? Is Poisson distribution applied on the training data for scVI and then the l1000_crispr data used for evaluation?

What I have been trying to do  in case of Geneformer is to take these steps:

1. Get pseudo-counts by de-normalizing the data -  `np.expm1(adata.X)` 
2. `sc.pp.normalize_total(adata, taraget_sum = 1e4, inplace=True)`
3. `sc.pp.log1p(adata)`
4. Then use n_counts = 1e4 for all cells to avoid the Geneformer's tokenizer from normalizing it again

After that I pass the matrix to GeneformerTokenizer (leaving target_sum=1e4).
This seems to work, but I’m not 100 % sure it matches the exact pipeline you used in the paper.

⸻

Could you confirm whether this is indeed the procedure you followed for the L1000 signatures?
If you used a different script—or have a notebook you can share—having that reference would make reproducing the results much easier. I’d be grateful for any pointers!

Thanks in advance, and great work on the benchmark.
